src/deps_log.cc

```diff
@@ -174,14 +173,24 @@ bool DepsLog::Load(const string& path, State* state, string* err) {
 
   bool valid_header = true;
   int version = 0;
-  if (!fgets(buf, sizeof(buf), f) || fread(&version, 4, 1, f) < 1)
+  if (!fgets(buf, sizeof(buf), f)) {
     valid_header = false;
+  } else if (strcmp(buf, kFileSignature) != 0) {
+    valid_header = false;
+  } else if (fread(&version, 4, 1, f) < 1) {
+    valid_header = false;
+  } else if (version != kCurrentVersion) {
+    valid_header = false;
+  } else if (fseek(f, -4, SEEK_CUR) != 0) {
+    valid_header = false;
+  } else if (fread(&buf, 1, 4, f) < 4) {
+    valid_header = false;
+  }
```

Comment:
IMHO, the original code was slightly more readable, I wouldn't change it (&see below the question about endianess handling).